docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -204,26 +204,443 @@ To make the actions more user-friendly you can also use alternating buttons as
+
+For example, a component expressed like this:
+```js
+<PhotoItem size={200} caption={'Angkor Wat'} onSelect={openLightbox}>
+	<img src="path/to/image.jpg" />
+</PhotoItem>
+```
 
-### SilverStripeComponent
+Might actually render HTML that looks like this:
+```html
+<div class="photo-item">
+	<div class="photo" style="width:200px;height:200px;">
+		<img src="path/to/image.jpg">
+	</div>
+	<div class="photo-caption">
+		<h3><a>Angkor Wat/a></h3>
+	</div> 
+</div>
+```
 
-The base class for SilverStripe React components. If you're building React components for the CMS, this is the class you want to extend. `SilverStripeComponent` extends `React.Component` and adds some handy CMS specific behaviour.
+This syntax is known as JSX. It is transpiled at build time into native Javascript calls
+to the React API. While optional, it is recommended to express components this way.
+
+This documentation will stop short of explaining React in-depth, as there is much better
+documentation available all over the web. We recommend:
+* [The Official React Tutorial](https://facebook.github.io/react/tutorial/tutorial.html)
+* [Build With React](http://buildwithreact.com/tutorial)
+
+#### A few words about ES6
+The remainder of this tutorial is written in [ECMAScript 6](http://es6-features.org/#Constants), or _ES6_
+for short. This is the new spec for Javascript (currently ES5) that is as of this writing
+only partially implmented in modern browsers. Because it doesn't yet enjoy vast native support,
+it has to be [transpiled](https://www.stevefenton.co.uk/2012/11/compiling-vs-transpiling/) in order to work
+in a browser. This transpiling can be done using a variety of toolchains, but the basic
+ principle is that a browser-ready, ES5 version of your code is generated in your dev
+ environment as part of your workflow.
+   
+   As stated above, there are many ways to solve the problem of transpiling. The toolchain
+   we use in core SilverStripe modules includes:
+   * [Babel](http://babeljs.io) (ES6 transpiler)
+   * [Webpack](http://webpack.js.org) (Module bundler)
+
+### Using dependencies within your component
+
+If your component has dependencies, you can add them via the injector using the `inject()`
+higher order component. The function accepts the following arguments:
+
+```js
+inject(Component, [dependencies], mapDependenciesToProps)
+```
+* **Component** The component definition to inject into
+* **[dependencies]**: An array of dependencies (or a string, if just one)
+* **mapDependenciesToProps**: (optional) All depdencies are passed into this function as params. The function
+is expected to return a map of props to dependencies. If this parameter is not specified,
+the prop names and the service names will mirror each other.
+
+__my-module/js/components/Gallery.js__
+```js
+import React from 'react';
+import { inject } from 'lib/Injector';
+
+class Gallery extends React.Component {
+  render() {
+    const { SearchComponent, ItemComponent } = this.props;
```

Comment:
Yeah, I see that. We need to show the utility of the mapping function to assign the services to arbitrary variables. It's the equivalent of `properties: { someVar: %$SomeService } ` in YAML. 

If left undefined, the mapping function assigns the props identical to the service name. I do think that's an important thing to show off, because it's less magic. Plus, it's more IDE friendly to declare explicitly the props you're using.

I think what I can do is show the example without the mapping function first, where the prop names magically mirror the service names, and then show an enhancement where they're mapped, which would be the "we recommend this" method.